todo-app/src/App.js

```diff
@@ -1,13 +1,98 @@
 import React, { Component } from 'react';
+import Task from './Task' 
+import TaskForm from './TaskForm'
+import {loadTodoItems, saveTodoItems} from './storage'
+
+
+export default class App extends Component {
+    state = {
+    todos: loadTodoItems(),
+    editingTodoID: null
+  }
+
+handleEditTodo = id => {
+  this.setState({editingTodoID: id});
+};
+
+handleCancleEdit = ()=> {
+    this.setState({editingTodoID: null});
+}
+
+
+setTodos(todos) {
+    this.setState({todos})
+    saveTodoItems(todos)
+}
+
+handleSaveEdit = (id , description) => {
+    this.setTodos(this.state.todos.map(todoItem => {
+        if (todoItem.id === id) {
+            this.setState({editingTodoID: null});
+            return {...todoItem, description};
+        }else {
+            return todoItem;
+        }
+    }))
+};
+toggleStatus = (todoID) =>{
+    this.setTodos(this.state.todos.map(todo =>{
+                if (todo.id === todoID){
+                    todo.done = !todo.done
```

Comment:
You are mutating your todo object directly here. Instead, you need to create a new object, like you do inside your `handleSaveEdit` function just above.